src/persist/src/indexed/columnar/arrow.rs

```diff
@@ -190,75 +182,62 @@ fn decode_arrow_file_kvtd<R: Read + Seek>(
     let projection = None;
     let file_reader = FileReader::new(r, file_meta, projection);
 
-    let file_schema = file_reader.schema().fields().as_slice();
+    let file_schema = file_reader.schema().fields.as_slice();
     // We're not trying to accept any sort of user created data, so be strict.
-    if file_schema != SCHEMA_ARROW_KVTD.fields() {
+    if file_schema != SCHEMA_ARROW_KVTD.fields {
         return Err(format!(
             "expected arrow schema {:?} got: {:?}",
-            SCHEMA_ARROW_KVTD.fields(),
-            file_schema
+            SCHEMA_ARROW_KVTD.fields, file_schema
         )
         .into());
     }
 
     let mut ret = Vec::new();
-    for batch in file_reader {
-        ret.push(decode_arrow_batch_kvtd(&batch?)?);
+    for chunk in file_reader {
+        ret.push(decode_arrow_batch_kvtd(&chunk?)?);
     }
     Ok(ret)
 }
 
-/// Converts a ColumnarRecords into an arrow [(K, V, T, D)] RecordBatch.
-pub fn encode_arrow_batch_kvtd(x: &ColumnarRecords) -> RecordBatch {
-    RecordBatch::try_new(
-        SCHEMA_ARROW_KVTD.clone(),
-        vec![
-            Arc::new(BinaryArray::from_data(
-                DataType::Binary,
-                x.key_offsets.clone(),
-                x.key_data.clone(),
-                None,
-            )),
-            Arc::new(BinaryArray::from_data(
-                DataType::Binary,
-                x.val_offsets.clone(),
-                x.val_data.clone(),
-                None,
-            )),
-            Arc::new(PrimitiveArray::from_data(
-                DataType::UInt64,
-                x.timestamps.clone(),
-                None,
-            )),
-            Arc::new(PrimitiveArray::from_data(
-                DataType::Int64,
-                x.diffs.clone(),
-                None,
-            )),
-        ],
-    )
+/// Converts a ColumnarRecords into an arrow [(K, V, T, D)] Chunk.
+pub fn encode_arrow_batch_kvtd(x: &ColumnarRecords) -> Chunk<Arc<dyn Array>> {
+    Chunk::try_new(vec![
+        Arc::new(BinaryArray::from_data(
+            DataType::Binary,
+            x.key_offsets.clone(),
+            x.key_data.clone(),
+            None,
+        )) as Arc<dyn Array>,
+        Arc::new(BinaryArray::from_data(
+            DataType::Binary,
+            x.val_offsets.clone(),
+            x.val_data.clone(),
+            None,
+        )),
+        Arc::new(PrimitiveArray::from_data(
+            DataType::UInt64,
+            x.timestamps.clone(),
+            None,
+        )),
+        Arc::new(PrimitiveArray::from_data(
+            DataType::Int64,
+            x.diffs.clone(),
+            None,
+        )),
+    ])
     .expect("schema matches fields")
 }
 
-/// Converts an arrow [(K, V, T, D)] RecordBatch into a ColumnarRecords.
-pub fn decode_arrow_batch_kvtd(x: &RecordBatch) -> Result<ColumnarRecords, String> {
-    // We're not trying to accept any sort of user created data, so be strict.
-    if x.schema().fields() != SCHEMA_ARROW_KVTD.fields() {
```

Comment:
bummer to lose the ability to have this check but I just looked and it's done at the file level in both places this is lose